### PR TITLE
[RFC] doc: Remove references to Windows 98

### PR DESCRIPTION
**Pull request by sethjackson**
Found some more references to Windows 98 (and 95).

I'm not sure if the fonts are shipped anymore but regardless we don't support Windows 95, 98 or NT anymore.

Also nuke a DOS reference in `runtime/doc/farsi.txt` while here and fix some grammar.

### CODE REVIEW

File: runtime/doc/mbyte.txt [hunk 1]
```diff
@@ -837,14 +837,11 @@ WHAT IS IME
...
-    On Windows 2000 and XP it should work as well (without downloading).  On
-    Windows 2000 Professional, Global IME is built in, and the Input Locales
-    can be added through Control Panel/Regional Options/Input Locales.
-    Please see below URL for detail of Global IME.  You can also find various
-    language version of Global IME at same place.
+    Global IME enables input of Chinese, Japanese, and Korean text into Vim
+    buffer on any language version of Windows.  Global IME is built in, and
+    the Input Locales can be added through Control Panel/Regional
+    Options/Input Locales.  Please see below URL for detail of Global IME.
+    You can also find various language version of Global IME at same place.
```
Comment (Pyrohh) [inline runtime/doc/mbyte.txt]: I'm pretty sure we can nuke large parts of this file, but that might be better left to a separate PR.

File: runtime/doc/farsi.txt [hunk 2]
```diff
@@ -61,15 +61,14 @@ directory:
 
    +  far-a01.pcf    X Windows fonts for Unix including Linux systems
    +  far-a01.bf     X Windows fonts for SunOS
-   +  far-a01.f16    a screen fonts for Unix including Linux systems
-   +  far-a01.fon    a monospaced fonts for Windows NT/95/98
-   +  far-a01.com    a screen fonts for DOS
+   +  far-a01.f16    screen fonts for Unix including Linux systems
+   +  far-a01.fon    monospaced fonts for Windows
```
Comment (Pyrohh) [inline runtime/doc/farsi.txt]: This whole `Farsi Fonts` section can be removed, as the fonts referenced above don't seem to exist in the Neovim repo.

Comment (sethjackson) [inline runtime/doc/farsi.txt]: :+1: I thought so as I couldn't find them either.

File: runtime/doc/mbyte.txt (hunk 1 above)
Comment (sethjackson) [inline runtime/doc/mbyte.txt]: Let's leave it for now. There are things still in mbyte.c and I think @equalsraf said there were some things in there to do for Windows support IIRC.

**Comment by sethjackson** [comment]
Also removed the font installation section since it refers to outdated Unix systems and fonts that we don't have anymore.

**Comment by Pyrohh** [comment]
Thanks @sethjackson.

File: runtime/doc/mbyte.txt (hunk 1 above)
Comment (justinmk) [inline runtime/doc/mbyte.txt]: We removed Vim's internal IME (OSes provide their own IME these days).

Comment (sethjackson) [inline runtime/doc/mbyte.txt]: :+1: cool. I'll probably open another PR to update the docs then.
